src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -494,6 +495,19 @@ QVariant QgsAttributeTableModel::headerData( int section, Qt::Orientation orient
       return tr( "feature id" );
     }
   }
+  else if ( role == Qt::DecorationRole && orientation == Qt::Vertical )
+  {
+    QgsRenderContext ctx;
+    layer()->rendererV2()->startRender( ctx, layer()->pendingFields() );
+    QgsFeatureId id = mRowIdMap[section];
+    QgsFeature feature;
+    mLayerCache->featureAtId( id, feature );
+    QgsSymbolV2List symbols = layer()->rendererV2()->symbolsForFeature( feature );
+    QgsSymbolV2* symbol = symbols.first();
```

Comment:
needs a check here that symbols isn't empty - otherwise crashes if the layer is using something like the heatmap renderer.